components/tikv_alloc/src/jemalloc.rs

```diff
@@ -30,13 +30,31 @@ pub fn fetch_stats() -> io::Result<Option<AllocStats>> {
     // Stats are cached. Need to advance epoch to refresh.
     JeEpoch::new()?.advance()?;
 
+    let mut dirty_page_cnt: c_uint = 0 as c_uint;
+    let mut c_uint_len: size_t = std::mem::size_of::<c_uint>() as size_t;
+    unsafe {
+        let _r = jemalloc_sys::mallctl(
+            &b"stats.arenas.4096.pdirty\0"[0] as *const _ as *const c_char,
+            &mut dirty_page_cnt as *mut _ as *mut c_void,
+            &mut c_uint_len as *mut _,
+            std::ptr::null_mut(),
+            0,
+        );
+    }
+
     Ok(Some(vec![
         ("allocated", stats::allocated()?),
         ("active", stats::active()?),
         ("metadata", stats::metadata()?),
         ("resident", stats::resident()?),
         ("mapped", stats::mapped()?),
         ("retained", stats::retained()?),
+        ("dirty", dirty_page_cnt as usize * 4096),
```

Comment:
I think you can calculate dirty pages by `resident` - `active` - `metadata`. 

Unused pages in jemalloc have a special state -- `muzzy`. I'm not sure whether `muzzy` pages occupy RSS since they're created by `madvie(MADV_FREE)`.